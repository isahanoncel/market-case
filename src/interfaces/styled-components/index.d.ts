// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft';

export type Colors =
  | 'primary'
  | 'secondary'
  | 'white'
  | 'black100'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
