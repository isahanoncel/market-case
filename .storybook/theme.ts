import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
