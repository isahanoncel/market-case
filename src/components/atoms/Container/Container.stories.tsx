import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Container from '.';

export default {
  title: 'Atoms/Container',
  component: Container,
} as ComponentMeta<typeof Container>;

export const Default: ComponentStory<typeof Container> = (args) => (
  <Container {...args}>Container</Container>
);
