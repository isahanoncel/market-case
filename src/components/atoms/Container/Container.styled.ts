import styled from 'styled-components';

export const ContainerWrapper = styled.div`
  max-width: 1232px;
  margin: 0 auto;
  height: 100%;
`;
