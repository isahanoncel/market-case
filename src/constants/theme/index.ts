import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
