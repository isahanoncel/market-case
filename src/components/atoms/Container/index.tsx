import React, { FC } from 'react';
import * as S from './Container.styled';

interface IContainer {
  children: React.ReactNode | string;
}
const Container: FC<IContainer> = ({
  children,
}) => {
  return (
    <S.ContainerWrapper>
      {children}
    </S.ContainerWrapper>
  );
};

export default Container;
