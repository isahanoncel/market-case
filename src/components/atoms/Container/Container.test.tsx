import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Container from '.';
import theme from '../../../constants/theme';

test('Container snapshot testing', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Container>Container</Container>
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
