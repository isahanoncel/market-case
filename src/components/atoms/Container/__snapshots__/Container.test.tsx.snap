// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container snapshot testing 1`] = `
.c0 {
  max-width: 1232px;
  margin: 0 auto;
  height: 100%;
}

<div
  className="c0"
>
  Container
</div>
`;
